enterprise/security/src/main/java/org/neo4j/server/security/enterprise/auth/plugin/api/AuthToken.java

```diff
@@ -19,20 +19,43 @@
  */
 package org.neo4j.server.security.enterprise.auth.plugin.api;
 
+import java.util.Map;
+
 /**
- * The predefined keys of the auth token <tt>Map&lt;String,Object&gt;</tt>.
+ * The authentication token provided by the client, which is used to authenticate the subject's identity.
+ *
+ * <p>A common scenario is to have principal be a username and credentials be a password.
+ *
+ * @see org.neo4j.server.security.enterprise.auth.plugin.spi.AuthenticationPlugin#authenticate(AuthToken)
+ * @see org.neo4j.server.security.enterprise.auth.plugin.spi.AuthPlugin#authenticateAndAuthorize(AuthToken)
  */
 public interface AuthToken
 {
-    String PRINCIPAL = "principal";
-    String CREDENTIALS = "credentials";
-    String REALM = "realm";
+    /**
+     * Returns the identity to authenticate.
+     *
+     * <p>Most commonly this is a username.
+     *
+     * @return the identity to authenticate.
+     */
+    String principal();
+
+    /**
+     * Returns the credentials that verifies the identity.
+     *
+     * <p>Most commonly this is a password.
+     *
+     * @return the credentials that verifies the identity.
+     */
+    char[] credentials();
 
     /**
-     * The corresponding value of this key is a <tt>Map<String,Object></tt> of custom parameters
-     * as provided by the client. This can be used as a vehicle to connect a client application
-     * with a server-side auth plugin.
-     * Neo4j will act as a pure transport and will not inspect the contents of this map.
+     * Returns an optional custom parameter map if provided by the client.
+     *
+     * <p>This can be used as a vehicle to send arbitrary auth data from a client application
+     * to a server-side auth plugin. Neo4j will act as a pure transport and will not touch the contents of this map.
+     *
+     * @return a custom parameter map if provided by the client, otherwise <tt>null</tt>
      */
-    String PARAMETERS = "parameters";
+    Map<String,Object> parameters();
```

Comment:
We could but would then also lose the distinction between if the map was provided by the client or if an empty map was provided (not sure if that is useful, but anyways..)